glusterd2/commands/snapshot/snapshot-create.go

```diff
@@ -656,22 +656,16 @@ func snapshotCreateHandler(w http.ResponseWriter, r *http.Request) {
 
 	txn, err := transaction.NewTxnWithLocks(ctx, req.VolName, req.SnapName)
 	if err != nil {
-		if err == transaction.ErrLockTimeout {
-			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
-		} else {
-			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
-		}
+		status, err := restutils.GetStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
 		return
 	}
 	defer txn.Done()
 
 	vol, e := volume.GetVolume(req.VolName)
 	if e != nil {
-		if e == gderrors.ErrVolNotFound {
-			restutils.SendHTTPError(ctx, w, http.StatusNotFound, gderrors.ErrVolNotFound)
-		} else {
-			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, e)
-		}
+		status, err := restutils.GetStatusCode(err)
```

Comment:
Incorrect variable is used here. The error variable here is named `e`. You have to change that to `err` and pass that